File: 0683/Sesión 05 - lib C#/Publicaciones_v01/Publicaciones_v01/Libro.cs
using Publicaciones;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones.EstandardIEEE
{
    public class Libro: Publicacion
    {
        private string edicion;
        private string lugar;
        private string editorial;

        public Libro(string nombre, string nombre_completo_autor, int año, string edicion, string lugar, string editorial) 
            : base(nombre, nombre_completo_autor, año)
        {
            this.edicion= edicion;
            this.lugar = lugar;
            this.editorial = editorial;
        }

        public Libro(string nombre, string[] nombres_completos_autores, int año, string edicion, string lugar, string editorial) 
            : base(nombre, nombres_completos_autores, año)
        {
            this.edicion = edicion;
            this.lugar = lugar;
            this.editorial = editorial;
        }

        public override string ToString()
        {
            string referencia = "["+ Publicacion.orden_presentacion + "] ";
            for (int i = 0; i < this.autores.Count; i++)
            {
                if (i != 0)
                {
                    if (i == this.cantidad_autores - 1)
                        referencia += " and ";
                    else
                        referencia += ", ";
                }
                referencia += this.autores[i].ToString();
            }
            referencia += ". ";
            referencia += this.nombre;
            referencia += ". ";
            referencia += this.edicion;
            referencia += ". ";
            referencia += this.lugar;
            referencia += ". ";
            referencia += this.editorial;
            referencia += ". ";
            referencia += this.año;
            referencia += ".";
            return referencia;
        }
    }
}

File: 0683/Sesión 04 - lib java/Publicaciones_v02/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones02/publicaciones/estandardAPA/Libro.java
package pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones.estandardAPA;

import pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones.Publicacion;

public class Libro extends Publicacion {

    private String edicion;
    private String lugar;
    private String editorial;

    public Libro(String nombre, String nombre_completo_autor, Integer año, String edicion, String lugar, String editorial) {
        super(nombre, nombre_completo_autor, año);
        this.edicion = edicion;
        this.lugar = lugar;
        this.editorial = editorial;
    }

    public Libro(String nombre, String nombres_completos_autores[], Integer año, String edicion, String lugar, String editorial) {
        super(nombre, nombres_completos_autores, año);
        this.edicion = edicion;
        this.lugar = lugar;
        this.editorial = editorial;
    }

    @Override
    public String toString() {
        String referencia = "";
        for (int i = 0; i < cantidad_autores; i++) {
            if (i != 0) {
                if (i == cantidad_autores - 1) {
                    referencia = referencia.concat(" and ");
                } else {
                    referencia = referencia.concat(", ");
                }
            }
            referencia = referencia.concat(autores.get(i).toString());
        }
        referencia = referencia.concat(". (");
        referencia = referencia.concat(año.toString());
        referencia = referencia.concat("). ");        
        referencia = referencia.concat(nombre);
        referencia = referencia.concat(". ");
        referencia = referencia.concat(edicion);
        referencia = referencia.concat(". ");
        referencia = referencia.concat(lugar);
        referencia = referencia.concat(". ");
        referencia = referencia.concat(editorial);
        referencia = referencia.concat(". ");
        return referencia;
    }
}